src/lib/comms/include/sol-network.h

```diff
@@ -234,31 +235,66 @@ static inline bool
 sol_network_link_addr_eq(const struct sol_network_link_addr *a,
     const struct sol_network_link_addr *b)
 {
-    const uint8_t *addr_a, *addr_b;
     size_t bytes;
 
-    if (a->family != b->family)
-        return false;
+    if (a->family == b->family) {
+        const uint8_t *addr_a, *addr_b;
+
+        if (a->family == SOL_NETWORK_FAMILY_INET) {
+            addr_a = a->addr.in;
+            addr_b = b->addr.in;
+            bytes = sizeof(a->addr.in);
+        } else if (a->family == SOL_NETWORK_FAMILY_INET6) {
+            addr_a = a->addr.in6;
+            addr_b = b->addr.in6;
+            bytes = sizeof(a->addr.in6);
+        } else if (a->family == SOL_NETWORK_FAMILY_BLUETOOTH) {
+            if (a->addr.bt_type != b->addr.bt_type)
+                return false;
+
+            addr_a = a->addr.bt_addr;
+            addr_b = b->addr.bt_addr;
+            bytes = sizeof(a->addr.bt_addr);
+        } else
+            return false;
+        return !memcmp(addr_a, addr_b, bytes);
+    }
+
+    if ((a->family == SOL_NETWORK_FAMILY_INET &&
+         b->family == SOL_NETWORK_FAMILY_INET6) ||
+        (a->family == SOL_NETWORK_FAMILY_INET6 &&
+         b->family == SOL_NETWORK_FAMILY_INET)) {
+
+        struct ipv6_map_prefix {
+            const uint8_t zeroes[10];
+            const uint16_t ones;
```

Comment:
Hmm... Maybe I'm missing something, but I guess `s6_addr32[0]` through `s6_addr32[2]` just captures the first 96 bits of the IPv6 address, the same as `uint8_t zeroes[10]` through `uint16_t ones`. The difference is that in `netinet/in.h`, since it's using "32-bit chunks", `htonl()` just pads the last 32-bit chunk with 16 `0` bits before those 16 `1` bits from `0xffff`. `ipv6_map_prefix` just uses an 80-bit chunk followed by a 16-bit chunk instead.

If that's true, both cases seems to be correctly checking for a 96-bit portion with 80 `0`s followed by 16 `1`s. But I guess @cabelitos can clear it out a little better. :)